packages/xo-server/src/xo-mixins/migrate-vm.mjs

```diff
@@ -0,0 +1,102 @@
+import { Backup } from '@xen-orchestra/backups/Backup.js'
+
+export default class VmMover {
+  constructor(app) {
+    this._app = app
+  }
+
+  async warmMigrateVm(sourceVmId, srId, startDestVm = true, deleteSource = false, { warmingCycle = 1 } = {}) {
+    // we'll use a one time use continuous replication job with the VM to migrate
+
+    const app = this._app
+    const sourceVm = app.getXapiObject(sourceVmId)
+    const config = {
+      snapshotNameLabelTpl: '[XO Lukewarm migration {job.name}] {vm.name_label}',
+    }
+    const job = {
+      type: 'backup',
+      id: 'import',
+      mode: 'delta',
+      vms: { id: sourceVmId },
+      name: `Lukewarm migration`,
+      srs: { id: srId },
+      settings: {
+        '': {
+          // mandatory for delta replication writer
+          copyRetention: 1,
+        },
+      },
+    }
+    const schedule = { id: 'one-time' }
+
+    // for now we only support this from the main OA, no proxy
+    const backup = new Backup({
+      config,
+      job,
+      schedule,
+      getAdapter: async remoteId => app.getBackupsRemoteAdapter(await app.getRemoteWithCredentials(remoteId)),
+
+      // `@xen-orchestra/backups/Backup` expect that `getConnectedRecord` returns a promise
+      getConnectedRecord: async (xapiType, uuid) => app.getXapiObject(uuid),
+    })
+    // run a few time to have a a warm target , as close as possible as the hot one running
+    for (let cycle = 0; cycle < warmingCycle; cycle++) {
+      await backup.run()
+    }
+    const xapi = sourceVm.$xapi
+    const ref = sourceVm.$ref
+
+    // stop the source VM before
+    try {
+      await xapi.callAsync('VM.clean_shutdown', ref)
+    } catch (error) {
+      await xapi.callAsync('VM.hard_shutdown', ref)
+    }
+    // make it so it can't be restarted by error
+    await sourceVm.update_blocked_operations({
+      start: 'Start operation for this vm is blocked, clone it if you want to use it.',
+      start_on: 'Start operation for this vm is blocked, clone it if you want to use it.',
```

Comment:
Clearer message: *This VM has been migrated somewhere else and might not be up to date, check twice before starting it*.